expr/cast.go

```diff
@@ -114,6 +116,20 @@ func castToTime(zv zng.Value) (zng.Value, error) {
 	return zng.Value{zng.TypeTime, zng.EncodeTime(nano.Ts(ns))}, nil
 }
 
+func castToDuration(zv zng.Value) (zng.Value, error) {
+	if zng.IsFloat(zv.Type.ID()) {
+		f, _ := zng.DecodeFloat64(zv.Bytes)
+		ts := int64(nano.FloatToTs(f))
+		// XXX GC
+		return zng.Value{zng.TypeDuration, zng.EncodeDuration(ts)}, nil
+	}
+	ns, ok := coerce.ToInt(zv)
+	if !ok {
+		return zng.Value{}, ErrBadCast
+	}
+	return zng.Value{zng.TypeDuration, zng.EncodeDuration(ns)}, nil
+}
+
```

Comment:
Nit: Follow the order established in `LookupPrimitiveCaster` by putting this before `castToTime`.